src/collectors/consul/ConsulCollector.py

```diff
@@ -0,0 +1,133 @@
+        try:
+            response = urllib2.urlopen(url + '/v1/catalog/nodes')
+            nodes = json.load(response)
+
+            self.publish('catalog.total_nodes', len(nodes))
+
+            # Get the health for each node
+            node_healths = map(self.get_node_health, nodes)
+            # Filter down to nodes with health responses
+            up_node_healths = [node for node in node_healths if node]
+            # Calculate the max status for the checks on each node
+            max_statuses = map(self.get_check_max_status, up_node_healths)
+
+            # Increment the metrics for how many nodes are of each status
+            self.publish('catalog.nodes_up', len(up_node_healths))
+            self.publish('catalog.nodes_critical', len(
+                [status for status in max_statuses if status == 'critical']))
+            self.publish('catalog.passing', len(
+                [status for status in max_statuses if status == 'passing']))
+            self.publish('catalog.warning', len(
+                [status for status in max_statuses if status == 'warning']))
+        except Exception, err:
+            self.log.error("%s: %s", url, err)
+            return False
+
+    def collect_service_metrics(self):
+        url = self.config['url']
+        try:
+            services = list(json.load(urllib2.urlopen(
+                url + '/v1/catalog/services')).keys())
+
+            service_status = {'up': 0, 'passing': 0,
+                              'warning': 0, 'critical': 0}
+
+            for service in services:
+                nodes_for_service = self.get_nodes_for_service(service)
+
+                # Find the maximum status for the service on each node
+                node_max_statuses = map(lambda node: self.get_check_max_status(
+                    node['Checks']), nodes_for_service)
+
+                # Calculate the max status of the service across nodes
+                service_max_status = self.get_max_status(node_max_statuses)
+
+                # Increment the count for the service being up and for it's status
+                service_status['up'] += 1
+                service_status[service_max_status] += 1
+
+            for key in service_status.keys():
+                self.publish('service.services_' + key, service_status[key])
+        except Exception, err:
+            self.log.error("%s: %s", url, err)
+            return False
+
+    # Collect agent metrics (https://www.consul.io/api/agent.html#view-metrics)
+    def collect_agent_metrics(self):
+        url = self.config['url']
+        try:
+            metrics = json.load(urllib2.urlopen(url + '/v1/agent/metrics'))
+
+            for gauge in metrics['Gauges']:
+                self.publish(self.trimmed_metric_name(
+                    gauge['Name']), gauge['Value'])
+
+            for counter in metrics['Counters']:
+                self.publish(
+                    'counter.' + self.trimmed_metric_name(counter['Name']), counter['Count'])
```

Comment:
`counters.` is added to make this easily capturable in an ACS config. All these metrics are counters.

Reply:
These metrics would come through as `consul.counter.foo`?

Reply:
Yes.